unitTests/HostMatrix.C

```diff
@@ -0,0 +1,212 @@
+}
+
+TEST(HostMatrix, canGenerateLocalSparsityPattern)
+{
+    auto hostMatrix = ((HostMatrixEnvironment *)global_env)->hostMatrix;
+    auto exec = ((HostMatrixEnvironment *)global_env)->exec;
+
+    auto localSparsity =
+        hostMatrix->compute_local_sparsity(exec->get_device_exec());
+    std::vector<label> rows_expected({0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3,
+                                      3, 3, 3, 4, 4, 4, 4, 4, 5, 5, 5,
+                                      5, 6, 6, 6, 7, 7, 7, 7, 8, 8, 8});
+    std::vector<label> cols_expected({0, 1, 3, 0, 1, 2, 4, 1, 2, 5, 0,
+                                      3, 4, 6, 1, 3, 4, 5, 7, 2, 4, 5,
+                                      8, 3, 6, 7, 4, 6, 7, 8, 5, 7, 8});
+
+    // we have 9x9 matrix with 33 nnz entries
+    EXPECT_EQ(localSparsity->size_, 33);
+    EXPECT_EQ(localSparsity->dim[0], 9);
+    EXPECT_EQ(localSparsity->dim[1], 9);
+
+    // since we don't have any processor interfaces we only have
+    // a single interface span ranging from 0 to 33
+    EXPECT_EQ(localSparsity->interface_spans.size(), 1);
+    EXPECT_EQ(localSparsity->interface_spans[0].begin, 0);
+    EXPECT_EQ(localSparsity->interface_spans[0].end, 33);
+
+    auto res_size{localSparsity->col_idxs.get_size()};
+    std::vector<label> rows_res(localSparsity->row_idxs.get_data(),
+                                localSparsity->row_idxs.get_data() + res_size);
+    std::vector<label> cols_res(localSparsity->col_idxs.get_data(),
+                                localSparsity->col_idxs.get_data() + res_size);
+
+
+    EXPECT_EQ(rows_expected, rows_res);
+    EXPECT_EQ(cols_expected, cols_res);
+}
+
+TEST(HostMatrix, canGenerateNonLocalSparsityPattern)
+{
+    auto hostMatrix = ((HostMatrixEnvironment *)global_env)->hostMatrix;
+    auto exec = ((HostMatrixEnvironment *)global_env)->exec;
+    auto comm = exec->get_gko_mpi_device_comm();
+
+    auto nonLocalSparsity =
+        hostMatrix->compute_non_local_sparsity(exec->get_device_exec());
+
+    // corresponds to cell ids
+    std::vector<std::vector<label>> rows_expected({{2, 5, 8, 6, 7, 8},
+                                                   {0, 3, 6, 6, 7, 8},
+                                                   {0, 1, 2, 2, 5, 8},
+                                                   {0, 1, 2, 0, 3, 6}});
+    // we dont test the cols expected for now,
+    // as they are in compressed format
+
+    EXPECT_EQ(nonLocalSparsity->interface_spans.size(), 2);
+
+    EXPECT_EQ(nonLocalSparsity->interface_spans[0].begin, 0);
+    EXPECT_EQ(nonLocalSparsity->interface_spans[0].end, 3);
+    EXPECT_EQ(nonLocalSparsity->interface_spans[1].begin, 3);
+    EXPECT_EQ(nonLocalSparsity->interface_spans[1].end, 6);
+
+    auto res_size{nonLocalSparsity->row_idxs.get_size()};
+    std::vector<label> rows_res(
+        nonLocalSparsity->row_idxs.get_data(),
+        nonLocalSparsity->row_idxs.get_data() + res_size);
+    EXPECT_EQ(rows_expected[comm->rank()], rows_res);
+}
+
```

Comment:
We may need some unit tests for the member functions of the class `HostMatrixWrapper`?